orchestrator/relayer/src/main_loop.rs

```diff
@@ -33,9 +44,30 @@ pub async fn relayer_main_loop(
     relayer_config: RelayerConfig,
 ) {
     let mut grpc_client = grpc_client;
+
+    let gas_entries = ALTRUISTIC_RELAY_PERIOD / relayer_config.relayer_loop_speed;
+    let mut gas_estimator = GasTracker::new(gas_entries as usize);
     loop {
         let loop_start = Instant::now();
 
+        // update the gas estimator and determine if we should relay altruistically
```

Comment:
My only suggestion here would be to only update (or maybe even create) a GasTracker if we are running an altruistic relayer.

Reply:
I did consider that, but it seemed like it would make the conditional tree deeper for not all that much performance gain, the gas price request is a light weight request.